### ISSUE TITLE
Some sites defeat the addon field attribute posting ability

### ISSUE THREAD

**Issue by wellread1** [erichgoldman/add-url-to-window-title#2]
In my limited testing I have noted two sites that seem able to defeat add-url-to-window-title's
ability to post the field attributes of the field that has the focus.  I am posting them in case you have an interest in investigating examples like this.

* The main login page of MINT <https://wwws.mint.com/login.event> displays the Window Title: 
"Mint Start Here | wwws.mint.com/ - Mozilla Firefox" without the quotes

* The security question page only of the multi-page USBank login which appears to be accessible via <https://www.usbank.com/index.html> (by entering a dummy string as a username) displays the Window Title: "Online Banking | onlinebanking.usbank.com/ - Mozilla Firefox" without the quotes

**Comment by erichgoldman** [comment]
I will investigate. Sometimes if the form is in a frame or iframe, or loaded by JavaScript it is not part of the  DOM and cannot be targeted,

**Comment by erichgoldman** [comment]
When visiting US Bank's home page at: https://www.usbank.com/index.html, I was able to replicate the problem, however, during testing the bank flagged for failed attempts while going back and forth. Using developer tools it showed that the focus event was properly attached, but its not clear if one of the native focus events was interfering. I was not able to figure it out before I got locked out. I will try again in a few days unless they flagged my IP and I can't get back to it,

**Comment by erichgoldman** [comment]
I was able to replicate this on the Mint login page. What seems to be happening is that something in Mint's JavaScript is removing the `eventListner` that the add-on puts in the page. As a result, it there is no event to fire when it has focus. I am having some difficulty going through Mint's JavaScript and determining what exactly is doing this.

As part of the testing, I visited the page. I then disabled and re-enabled the add-on. When you re-enable the login (after the Mint page has fully loaded) then the field attributes will show. 

The add-on is added based on `pageshow` and not on `ready` otherwise the way that FireFox loads preferences won't be updated properly. Without introducing some periodic time based check (which would hurt general performance) it may not be possible to catch it in this case.

The combination of JavaScript and other factors baked into the native page means that sometimes the add-on will not work for that site. I am going to leave this open and will periodically see if I can figure out some alternatives or if someone else may be able to find a solution. Please continue to post any other sites you come across with issues.

**Comment by wellread1** [comment]
The Yahoo *pop-up* login dialog generated via the my.yahoo.com sign-in has the same problem.  However if the URL of the pop-up is pasted into the Firefox URL address bar the URL-in-Title works fine.

**Comment by erichgoldman** [comment]
If the HTML for that form is generated after the page has loaded or its in an iframe, the `focus` event will never be attached. I will look into the performance impact of adding a more general mutation observer in the case of later JavaScript. Perhaps if its a trade off between performance and the number of sites on which it will work that can be a user selectable option. 

I need to develop a procedure, but I think for now, each site where the add-on doesn't function as desired should be its own issue that way it will be easier to troubleshoot, research, and close issues.  

I appreciate all the feedback and reports either way!

**Comment by erichgoldman** [comment]
> The add-on is added based on pageshow and not on ready otherwise the way that FireFox loads preferences won't be updated properly.

I was mistaken, per the documentation `pageshow` should be firing after `load` and `read`

 - https://groups.google.com/forum/#!msg/mozilla-labs-jetpack/kDRvNZaWjR8/r90DF1I-64gJ
 - https://developer.mozilla.org/en-US/Add-ons/SDK/High-Level_APIs/tabs#open(options)

I will continue tracking why this does not open.

**Comment by erichgoldman** [comment]
I think I have debugged the first two issues:
For Mint, in my dev environment I have repackaged with the current version of the cfx compiler, which seems to have address.

For the US Bank example, the username field has no "name" attribute, only id. Near the end of development I had added a [sanitization check](https://github.com/erichgoldman/add-url-to-window-title/blob/master/data/add-url-to-title.js#L263), but it looks like I did not have any test cases where a string being passed for sanitization was null. I have added in a check on the parameter to the function to ensure its a string or that its set to an empty try `''`. I may run a few more tests against this before updating git and Mozilla.

For the Yahoo issue, I noticed that the url bar is grayed out. It looks like they launch the window with restrictions. I will need to investigate how this works and the impact on the add-on. However, for the time being the regular login that doesn't launch in a new window seems to work as an alternative.
